Cthulhu assumes arguments are valid identifiers

This is not true:
```
julia> f(var"#"::Int) = 1
f (generic function with 1 method)

julia> @descend f(1)
ERROR: AssertionError: kind(arg) == K"Identifier" || is_operator(arg)
Stacktrace:
  [1] (::TypedSyntax.var"#argidentifier#12")(arg::JuliaSyntax.TreeNode{TypedSyntax.TypedSyntaxData})
    @ TypedSyntax ~/.julia/packages/TypedSyntax/cH1Nu/src/node.jl:146
  [2] map_signature!(sig::JuliaSyntax.TreeNode{TypedSyntax.TypedSyntaxData}, slotnames::Vector{Symbol}, slottypes::Vector{Any})
    @ TypedSyntax ~/.julia/packages/TypedSyntax/cH1Nu/src/node.jl:170
  [3] map_signature!
    @ ~/.julia/packages/TypedSyntax/cH1Nu/src/node.jl:118 [inlined]
  [4] JuliaSyntax.TreeNode{TypedSyntax.TypedSyntaxData}(rootnode::JuliaSyntax.SyntaxNode, src::Core.CodeInfo, mappings::Vector{Vector{Union{…}}}, symtyps::IdDict{JuliaSyntax.SyntaxNode, Any})
    @ TypedSyntax ~/.julia/packages/TypedSyntax/cH1Nu/src/node.jl:86
  [5] tsn_and_mappings(mi::Core.MethodInstance, src::Core.CodeInfo, rt::Any, sourcetext::SubString{String}, lineno::Int32; warn::Bool, strip_macros::Bool, kwargs::@Kwargs{})
    @ TypedSyntax ~/.julia/packages/TypedSyntax/cH1Nu/src/node.jl:55
  [6] tsn_and_mappings(mi::Core.MethodInstance, src::Core.CodeInfo, rt::Any; warn::Bool, strip_macros::Bool, kwargs::@Kwargs{})
    @ TypedSyntax ~/.julia/packages/TypedSyntax/cH1Nu/src/node.jl:39
  [7] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/reflection.jl:361
  [8] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/reflection.jl:360
  [9] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/reflection.jl:37
 [10] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:496
 [11] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:397
 [12] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:791
 [13] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:791
 [14] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:807
 [15] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:804
 [16] #__descend_with_error_handling#89
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:222
 [17] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:211
 [18] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:203
 [19] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:169
 [20] top-level scope
    @ ~/.julia/packages/Cthulhu/8y2RA/src/Cthulhu.jl:169
 [21] top-level scope
    @ REPL[45]:1
Some type information was truncated. Use `show(err)` to see complete types.
```

--- a/TypedSyntax/src/node.jl
+++ b/TypedSyntax/src/node.jl
@@ -142,6 +142,10 @@ function map_signature!(sig::TypedSyntaxNode, slotnames::Vector{Symbol}, slottyp
         if kind(arg) == K"curly"
             arg = first(children(arg))
         end
+        if kind(arg) == K"var"
+            @show arg children(arg)
+            arg = child(arg, 1)
+        end
         kind(arg) == K"tuple" && return nothing, defaultval     # FIXME? see extrema2 test
         @assert kind(arg) == K"Identifier" || is_operator(arg)
         return arg, defaultval

--- a/TypedSyntax/src/node.jl
+++ b/TypedSyntax/src/node.jl
@@ -143,7 +143,6 @@ function map_signature!(sig::TypedSyntaxNode, slotnames::Vector{Symbol}, slottyp
             arg = first(children(arg))
         end
         if kind(arg) == K"var"
-            @show arg children(arg)
             arg = child(arg, 1)
         end
         kind(arg) == K"tuple" && return nothing, defaultval     # FIXME? see extrema2 test

--- a/TypedSyntax/test/runtests.jl
+++ b/TypedSyntax/test/runtests.jl
@@ -687,6 +687,12 @@ include("test_module.jl")
     @test_nowarn str = sprint(tsn; context=:color=>false) do io, obj
         printstyled(io, obj; hide_type_stable=false)
     end
+
+    # issue 597
+    tsn = TypedSyntaxNode(TSN.f597, (Int,))
+    @test_nowarn str = sprint(tsn; context=:color=>false) do io, obj
+        printstyled(io, obj; hide_type_stable=false)
+    end
 end
 
 if parse(Bool, get(ENV, "CI", "false"))

--- a/TypedSyntax/test/test_module.jl
+++ b/TypedSyntax/test/test_module.jl
@@ -247,6 +247,8 @@ function obfuscated(x)
     return f(x)
 end
 
+f597(var"#"::Int) = 1
+
 module Internal
 export helper
 helper(x) = x+1

